packages/desktop-client/src/components/reports/ReportOptions.ts

```diff
@@ -150,7 +150,15 @@ export const categoryLists = (categories: {
   grouped: CategoryGroupEntity[];
 }) => {
   const categoryList = [
-    ...categories.list,
+    ...categories.list.sort((a, b) => {
+      const catGroupA = categories.grouped.find(f => f.id === a.cat_group);
+      const catGroupB = categories.grouped.find(f => f.id === b.cat_group);
+      return a.sort_order && b.sort_order && catGroupA && catGroupB
```

Comment:
suggestion: Can we comment the return to better describe the condition going on here. It's a pretty intense condition to just read later without context as to why.